### PR TITLE
Adding LDDL support to Bert

### PR DESCRIPTION
**Pull request by wdykas**
# What does this PR do ?

Add LDDL data loader support to Megatron Bert
**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/conf/megatron_bert_config.yaml [hunk 1]
```diff
@@ -132,8 +132,8 @@ model:
     splits_string: 900,50,50
     seq_length: ${model.encoder_seq_length}
     skip_warmup: True
-    num_workers: 0
-    dataloader_type: single # cyclic
+    num_workers: 1
```
Comment (shanmugamr1992) [inline examples/nlp/language_modeling/conf/megatron_bert_config.yaml:135]: Is this a fixed value ?

File: examples/nlp/language_modeling/run_LDDL_bert.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+HYDRA_FULL_ERROR=1 torchrun --nnode 1 --nproc_per_node 4 megatron_bert_pretraining.py \
```
Comment (shanmugamr1992) [inline examples/nlp/language_modeling/run_LDDL_bert.sh:1]: Can we also add some README , on why use LDDL and when to use it ?

Comment (wdykas) [reply examples/nlp/language_modeling/run_LDDL_bert.sh:1]: Sorry I wasn't planning on actually commiting this script.  I was just using it for testing.  I can add a readme on using LDDL

File: examples/nlp/language_modeling/conf/megatron_bert_config.yaml (hunk 1 above)
Comment (wdykas) [reply examples/nlp/language_modeling/conf/megatron_bert_config.yaml:135]: I will put the config back to they way it was but add an LDDL option in the comment next to dataloader type.  num workers is the same as in normal pytorch it is a dataloader argument https://github.com/NVIDIA/NeMo/pull/5977/files#diff-d496e911eb06fca3128061957d326a83ba0a603ca8482ad292aee3851f49bf61R515

**Pull request closed without merging**